### PR TITLE
[cli] Make preflight test containers with unique name

### PR DESCRIPTION
**Pull request by TylerJewell**
### What does this PR do?
Solves race condition when starting multiple Che servers quickly where the preflight test containers were sharing the same name and running into conflicts. This sets the name of preflight containers to be unique by appending the port of the Che server.

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/4353

#### Changelog
[cli] Give containers used in preflight tests unique name
 
#### Release Notes
N/A

#### Docs PR
N/A

### CODE REVIEW

File: dockerfiles/base/scripts/base/library.sh [hunk 1]
```diff
@@ -256,21 +256,28 @@ docker_compose() {
   fi
 }
 
+# Return srandom character alphanumeric string (upper and lower case) of length $1
+generate_random_string() {
```
Comment (benoitf) [inline dockerfiles/base/scripts/base/library.sh]: I don't see where this function is used ?

Comment (TylerJewell) [reply dockerfiles/base/scripts/base/library.sh]: It's not anymore.  My first commit was going to use it with a 10 character code.  And then I decided to use `CHE_PORT` instead as the unique ID.  But I wanted to leave this utility method in the library for future use.

Comment (benoitf) [reply dockerfiles/base/scripts/base/library.sh]: could we only add it if we require it ?

Comment (TylerJewell) [reply dockerfiles/base/scripts/base/library.sh]: I am sure that is typical, but I can see a number of situations in the near future where we may want to use it.  It wasn't the easiest thing to dig up - so is it inappropriate to leave in unused methods?

Comment (benoitf) [reply dockerfiles/base/scripts/base/library.sh]: yes I would say that we only let code really used in master. Dead code is usually deleted.
you may add the code to the pull request so if you need to find it again, you could get it.
Or paste it to a gist

Comment (TylerJewell) [reply dockerfiles/base/scripts/base/library.sh]: ok

```
# Return srandom character alphanumeric string (upper and lower case) of length $1
generate_random_string() {
  NEW_UUID=$(cat /dev/urandom | tr -dc 'a-zA-Z0-9' | fold -w $1 | head -n 1)
  echo $NEW_UUID
}
```

**Review by benoitf — APPROVED** [review]

**Pull request merged**